tmt/utils/__init__.py

```diff
@@ -78,6 +78,34 @@
     from tmt._compat.typing import Self, TypeAlias
 
 
+def _normalize_user_data(
```

Comment:
By moving into `tmt.utils`:

* the function is no longer "private", used by `artemis` plugin and no one else, therefore it should no longer start with an underscore,
* "user data" is a concept understood by Artemis and therefore by `artemis`, but it does not mean much to the rest of the code base. I would suggest finding a different name and adding a docstring. It was clear what was happening when it was in `tmt.steps.provision.artemis`, but what is `tmt.utils._normalize_user_data` doing? tmt does not accept any "user data"...